core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -199,12 +201,28 @@ public void offer(RewriteFileGroup group) {
       commitReadyCommitGroups();
     }
 
+    /**
```

Comment:
This method confuses me a bit,

1. It's called failedRewrite but sounds like its only for a very specific case of failure
2. It will never actually succeed since if it is called when service is closed it will fail the precondition and throw an exception

Reply:
Thanks RussellSpitzer,
It has been modified here, using szehon-ho's suggestion above.